### PR TITLE
fix: environmentSecretsName for kratos courier 

### PR DESCRIPTION
**Pull request by bkomraz1**
## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [x] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added the necessary documentation within the code base (if appropriate).

## Further comments

This pull requests adds support of environmentSecretsName for karpenter-courier and fixing following problem:

helm chart kratos does not implement loading environment variable from file for courier #588

### CODE REVIEW

**Comment by bkomraz1** [comment]
Hi @Demonsthere,

Can you please review the pull request?

File: helm/charts/kratos/templates/statefulset-mail.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@
 {{- $labels := ternary .Values.statefulSet.labels .Values.deployment.labels (not (empty .Values.statefulSet.labels )) -}}
 {{- $extraVolumeMounts := ternary .Values.statefulSet.extraVolumeMounts .Values.deployment.extraVolumeMounts (not (empty .Values.statefulSet.extraVolumeMounts )) -}}
 {{- $extraEnv := ternary .Values.statefulSet.extraEnv .Values.deployment.extraEnv (not (empty .Values.statefulSet.extraEnv )) -}}
+{{- $environmentSecretsName := ternary .Values.statefulSet.environmentSecretsName .Values.deployment.environmentSecretsName (not (empty .Values.statefulSet.environmentSecretsName )) -}}
```
Comment (Demonsthere) [inline helm/charts/kratos/templates/statefulset-mail.yaml:7]: IF this value is supposed to be defined only for the courier, then we don't need this part and can use it like `.Values.statefulSet.environmentSecretsName`

File: helm/charts/kratos/values.yaml [hunk 2]
```diff
@@ -389,6 +389,13 @@ statefulSet:
   #      lines, adjust them as necessary, and remove the curly braces after 'annotations:'.
   #      e.g.  sidecar.istio.io/rewriteAppHTTPProbers: "true"
 
+  # -- The secret specified here will be used to load environment variables with envFrom.
+  # This allows arbitrary environment variables to be provided to the application which is useful for
+  # sensitive values which should not be in a configMap.
+  # This secret is not created by the helm chart and must already exist in the namespace.
+  # https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#configure-all-key-value-pairs-in-a-secret-as-container-environment-variables
+  # environmentSecretsName:
```
Comment (Demonsthere) [inline helm/charts/kratos/values.yaml:397]: the value can be kept in the default values as `""` or `null` to make it more verbose. 
Please add a test, in the https://github.com/ory/k8s/tree/master/hacks/manifests dir add an example secret to be loaded and in the https://github.com/ory/k8s/blob/master/hacks/values/kratos.yaml set the value to point to that secret :)

File: helm/charts/kratos/templates/statefulset-mail.yaml (hunk 1 above)
Comment (bkomraz1) [reply helm/charts/kratos/templates/statefulset-mail.yaml:7]: It is not dedicated to be only defined for statefulSet. The main goal is reusing environments of the the deployment. But above code allow to define different environment for  statefulSet.

File: helm/charts/kratos/values.yaml (hunk 2 above)
Comment (bkomraz1) [reply helm/charts/kratos/values.yaml:397]: hacks/manifests/env-secrets.yaml: https://github.com/ory/k8s/pull/589/files#diff-33a5a5786a496aa6051cdcbbe89bbb8a204fe0ede2f9ade7ba6d5754311735df

hacks/values/kratos.yaml: https://github.com/ory/k8s/pull/589/files#diff-589731bd7b6a30a19645b4259c330b77fc6e5003b66fe9692f13fdef8d09d4cfR237

environmentSecretsName for StatefullSet is added to default values file in the same way as environmentSecretsName for Deployment is present

**Review by Demonsthere — APPROVED** [review]
Great, thanks :)

**Pull request merged**